cli/src/commands/parse.rs

```diff
@@ -17,6 +17,19 @@ pub struct ParsedLockfile {
     pub packages: Vec<PackageDescriptor>,
 }
 
+impl From<ParsedLockfile> for Vec<PackageDescriptorAndLockfilePath> {
```

Comment:
It looks like all the places that use this immediately call `.extend()`

If that's the use case, you could instead `impl IntoIterator for ParsedLockfile` and then just call `.extend(parsed_lockfile)` without creating a `Vec` in between